### PR TITLE
Ruby solr

### PR DESCRIPTION
**Pull request by jduss4**
Implements existing XSLT solr transformation with Ruby instead for greater flexibility

First large difference, reads through all documents before transformations begin and inserts
document ids into caseid files related to them, so this way caseids are no longer a stopping
point for being able to post only a single file at a time -- now each file has the information it
needs to run "solo" rather than having to run all the files in a particular order

Otherwise, splits code into personography, documents, and caseids although there is some overlap
for the sake of being able to see what each document type needs in terms of special treatment and fields

### CODE REVIEW

**Comment by karindalziel** [comment]
One thing to point out, which I don't think should stop pushing this: the dev site using this index has one fewer people than the live site. This may be because of a data change, I just wanted to point it out. 

Screen Shot 2019-03-14 at 3 00 14 PM

**Comment by jduss4** [comment]
I investigated this and the extra person is `per.003269` "Janney, Jacob" whose entry in the persons file is commented out with the note "person should not have been encoded."

I am choosing, therefore, not to worry about it :)  Hooray!

File: scripts/overrides/tei_to_solr_caseid.rb [hunk 1]
```diff
@@ -99,6 +98,14 @@ def documents(x)
     people_role.each do |role, people|
       build_person_fields(x, role, people)
     end
+
+    # now figure out which document had the most recent date and use that as the
+    # caseid date (and dateDisplay)
+    dates = dates.compact.reject(&:empty?).map{|d| CommonXml.date_standardize(d)}
+    if dates.length > 0
+      x.field(dates.last, "name" => "date")
```
Comment (techgique) [inline scripts/overrides/tei_to_solr_caseid.rb]: Does `dates.compact...` sort them based on their values? I don't think it does so I'm worried that using `dates.last` here isn't actually getting the most recent date

Comment (techgique) [reply scripts/overrides/tei_to_solr_caseid.rb]: Discussed with Jess and she'll address in a commit to master :+1:

**Pull request merged**